modules/launchd/default.nix

```diff
@@ -0,0 +1,179 @@
+  };
+
+  config = mkMerge [
+    {
+      assertions = [{
+        assertion = agentPlists != { } -> pkgs.stdenv.isDarwin;
+        message = let names = lib.concatStringsSep ", " (attrNames agentPlists);
+        in "Must use Darwin for modules that require Launchd: " + names;
+      }];
+    }
+
+    (mkIf isDarwin {
+      home.extraBuilderCommands = ''
+        ln -s "${agentsDrv}" $out/LaunchAgents
+      '';
+
+      home.activation.checkLaunchAgents =
+        hm.dag.entryBefore [ "writeBoundary" ] ''
+          checkLaunchAgents() {
+            local oldDir="$(readlink -m "$oldGenPath/LaunchAgents")"
+            local newDir=${escapeShellArg agentsDrv}
+            local dstDir=${escapeShellArg dstDir}
+
+            local oldSrcPath newSrcPath dstPath agentFile agentName
+
+            find -L "$newDir" -maxdepth 1 -name '*.plist' -type f -print0 \
+                | while IFS= read -rd "" newSrcPath; do
+              agentFile="''${newSrcPath##*/}"
+              agentName="''${agentFile%.plist}"
+              dstPath="$dstDir/$agentFile"
+              oldSrcPath="$oldDir/$agentFile"
+
+              if [[ ! -e "$dstPath" ]]; then
+                continue
+              fi
+
+              if ! cmp --quiet "$oldSrcPath" "$dstPath"; then
+                errorEcho "Existing file '$dstPath' is in the way of '$newSrcPath'"
+                exit 1
+              fi
+            done
+          }
+
+          checkLaunchAgents
+        '';
+
+      # NOTE: Launch Agent configurations can't be symlinked from the Nix store
+      # because it needs to be owned by the user running it.
+      home.activation.setupLaunchAgents =
+        hm.dag.entryAfter [ "writeBoundary" ] ''
+          setupLaunchAgents() {
+            local oldDir="$(readlink -m "$oldGenPath/LaunchAgents")"
+            local newDir="$(readlink -m "$newGenPath/LaunchAgents")"
+            local dstDir=${escapeShellArg dstDir}
+            local domain="gui/$UID"
+            local err=0
+
+            local srcPath dstPath agentFile agentName
+
+            find -L "$newDir" -maxdepth 1 -name '*.plist' -type f -print0 \
+                | while IFS= read -rd "" srcPath; do
+              agentFile="''${srcPath##*/}"
+              agentName="''${agentFile%.plist}"
+              dstPath="$dstDir/$agentFile"
+
+              if cmp --quiet "$srcPath" "$dstPath"; then
+                continue
+              fi
+              if [[ -f "$dstPath" ]]; then
+                while true; do
+                  $DRY_RUN_CMD launchctl bootout "$domain/$agentName" || err=$?
+                  if [[ -v DRY_RUN ]]; then
+                    break
+                  fi
+                  if (( err != 9216 )) &&
+                    ! launchctl print "$domain/$agentName" &> /dev/null; then
+                    sleep 1
+                    break
+                  fi
+                done
+              fi
+              $DRY_RUN_CMD install -Dm644 -T "$srcPath" "$dstPath"
```

Comment:
We should only make it readable, not writable + readable for the current user.

Reply:
Changed to `444` instead.